src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -1258,3 +1297,10 @@ bool clang_cpp_convertert::need_new_object(const clang::Stmt *parentStmt)
 
   return false;
 }
+
+symbolt *clang_cpp_convertert::get_fd_symbol(const clang::FunctionDecl &fd)
+{
+  std::string id, name;
+  get_decl_name(fd, name, id);
+  return (context.find_symbol(id));
```

Comment:
what if this fails?

Reply:
This function gives max freedom to the caller. 
It can return nullptr if fd symbol does not exist. But that would be a decision for the caller to make...
something like: 
```cpp
if(get_fd_symbol)
{
  // do_something if fd symbol exists
}
else
{
  // do_something if fd symbol doesn't exist
}
```

sometimes we just need the `if` block.

Reply:
In that case the documentation should say that `nullptr` is a valid result.